src/commands/database-get.ts

```diff
@@ -91,70 +100,45 @@ export default new Command("database:get <path>")
+      responseType: "stream",
+      resolveOnHTTPError: true,
     });
 
-    dbUrl = urlObj.href;
-
-    logger.debug("Query URL: ", dbUrl);
-    const reqOptions = {
-      url: dbUrl,
-    };
+    const fileOut = !!options.output;
+    const outStream = fileOut ? fs.createWriteStream(options.output) : process.stdout;
 
-    return api.addRequestHeaders(reqOptions).then((reqOptionsWithToken) => {
-      return new Promise((resolve, reject) => {
-        const fileOut = !!options.output;
-        const outStream = fileOut ? fs.createWriteStream(options.output) : process.stdout;
-        const writeOut = (s: Buffer | string, cb?: Function): void => {
-          if (outStream === process.stdout) {
-            outStream.write(s, cb);
-          } else if (outStream instanceof fs.WriteStream) {
-            outStream.write(s, (err) => {
-              if (cb) {
-                cb(err);
-              }
-            });
-          }
-        };
-        let erroring = false;
-        let errorResponse = "";
-        let response: any;
+    if (res.status >= 400) {
+      const r = await utils.streamToString(res.body);
+      let d;
+      try {
+        d = JSON.parse(r);
+      } catch (e) {
+        throw new FirebaseError("Malformed JSON response", { original: e, exit: 2 });
+      }
+      throw responseToError({ statusCode: res.status }, d);
+    }
 
-        request
-          .get(reqOptionsWithToken)
-          .on("response", (res) => {
-            response = res;
-            if (response.statusCode >= 400) {
-              erroring = true;
-            }
-          })
-          .on("data", (chunk) => {
-            if (erroring) {
-              errorResponse += chunk;
-            } else {
-              writeOut(chunk);
-            }
-          })
-          .on("end", () => {
-            writeOut("\n", () => {
-              resolve();
-            });
-            if (erroring) {
-              try {
-                const data = JSON.parse(errorResponse);
-                return reject(responseToError(response, data));
-              } catch (e) {
-                return reject(
-                  new FirebaseError("Malformed JSON response", {
-                    exit: 2,
-                    original: e,
-                  })
-                );
-              }
-            }
-          })
-          .on("error", reject);
-      });
+    res.body.pipe(outStream);
+    // Tack on a single newline at the end of the stream.
+    res.body.once("end", () => {
+      if (outStream === process.stdout) {
+        // `stdout` can simply be written to.
+        outStream.write("\n");
+      } else if (outStream instanceof fs.WriteStream) {
+        // .pipe closes the output file stream, so we need to re-open the file.
+        const s = fs.createWriteStream(options.output, { flags: "a" });
+        s.write("\n");
+        s.close();
+      } else {
+        throw new FirebaseError("Could not write line break", { status: 2 });
```

Comment:
Is this severe enough that we want to throw? What if we just let it go?

Reply:
Ehhh - probably not. I'll logger.debug it for posterity, but yeah, it wouldn't be the end of the world